lib/main.dart

```diff
@@ -310,16 +299,55 @@ enum VisualizerType {
   frequency,
 }
 
+class SoundSource {
+  String name;
+  String _path;
+
+  SoundSource.fromFilePath(this.name, this._path);
+
+  RandomAccessFile openForReading() {
+    return File(_path).openSync();
+  }
+}
+
 class _MyHomePageState extends State<MyHomePage> {
   VisualizerType _visualizerType = VisualizerType.time;
+  SoundSource source;
+  WavFile wavFile;
+
+  List<SoundSource> soundSources = [
+    SoundSource.fromFilePath("Piano C4",
+        '/Users/eseidel/Projects/rubberduck/sound/examples/third_party/freewavesamples.com/Casio-MT-45-Piano-C4.wav'),
+    SoundSource.fromFilePath("Guitar",
+        '/Users/eseidel/Projects/rubberduck/sound/examples/example.wav'),
+  ];
+
+  @override
+  void initState() {
+    setSource(soundSources.first);
+    super.initState();
```

Comment:
Consider calling the superclass at the beginning of the function rather than the end.